### PR TITLE
Moving to make our own load generator

### PR DESCRIPTION
**Pull request by Kn99HN**
# Description
- Update benchmark script to have both fortio and our own load generator
- Results from load generator will be transformed into buckets with counts and percentiles, similar to how fortio is creating the resulting data
- Changing from cpp to rust filters

### CODE REVIEW

**Comment by fruffy** [comment]
This will also have to be rebased.

File: kubernetes_env/benchmark/benchmark.py [hunk 1]
```diff
@@ -97,38 +134,75 @@ def plot(files):
...
-    cmd += f"http://{gateway_url}/productpage"
+    cmd += f"{url}"
     with open(output_file, "w") as f:
         fortio_res = util.exec_process(cmd,
                                        stdout=subprocess.PIPE,
                                        stderr=subprocess.PIPE)
         return fortio_res
 
 
+def burst_loop(url, threads, qps, run_time, queue):
```
Comment (fruffy) [inline kubernetes_env/benchmark/benchmark.py:166]: Hmm we should probably take a look at the traffic this one causes. For example with Grafana. That is pretty easy.

File: kubernetes_env/benchmark/benchmark.py [hunk 2]
```diff
@@ -143,22 +217,32 @@ def start_benchmark(fortio, filter_dirs, platform, threads, qps, time):
...
+                log.error(f"Error benchmarking for {fd}")
+                return util.EXIT_FAILURE
+        else:
+            log.info("Generating load...")
+            burst_res = do_burst(product_url, platform, threads, qps, time)
+            results.append(burst_res)
 
-    dataf = [f for f in DATA_DIR.glob("*") if f.is_file()]
-    return plot(dataf)
+    if fortio is True:
```
Comment (fruffy) [inline kubernetes_env/benchmark/benchmark.py]: Why not just if fortio?

File: kubernetes_env/benchmark/benchmark.py [hunk 3]
```diff
@@ -226,7 +309,7 @@ def main(args):
     parser.add_argument("-fo",
                         "--fortio",
                         dest="fortio",
-                        type=bool,
+                        default=True,
```
Comment (fruffy) [inline kubernetes_env/benchmark/benchmark.py]: This is a little counter intuitive. I would use a different flag if you want to use a custom load generator.  E.g, --use-custom or something like that.

File: kubernetes_env/benchmark/benchmark.py (hunk 1 above)
Comment (Kn99HN) [reply kubernetes_env/benchmark/benchmark.py:166]: This one: https://github.com/grafana/grafana?

**Pull request closed without merging**